src/main/java/io/cdap/plugin/topn/TopNConfig.java

```diff
@@ -0,0 +1,131 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.topn;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginConfig;
+import io.cdap.cdap.etl.api.FailureCollector;
+
+import javax.annotation.Nullable;
+
+/**
+ * Config for top N of plugins.
+ */
+public class TopNConfig extends PluginConfig {
+
+  public static final int MAX_TOP = 1024*1024;
+  public static final String FIELD = "field";
+  public static final String SIZE = "size";
+  public static final String IGNORE_NULL = "ignoreNull";
+
+  @Name(FIELD)
+  @Description("The field which top results are sorted by.")
+  @Macro
+  private String field;
+
+  @Name(SIZE)
+  @Description("The maximum number of top records sorted by 'field' in result (DEFAULT: 1)")
+  @Macro
+  @Nullable
+  private Integer size;
+
+  @Name(IGNORE_NULL)
+  @Description("Set to 'true' to ignore records with null value in the field to sort by (DEFAULT : false)")
+  @Macro
+  @Nullable
+  private Boolean ignoreNull;
+
+  public TopNConfig() {
+    this.field = "";
+    this.size = null;
+    this.ignoreNull = null;
+  }
+
+  public void validate(FailureCollector failureCollector, @Nullable Schema inputSchema) {
+
+    // Check existence and type of the field.
+    if (!containsMacro(FIELD)) {
+      Schema.Field field = inputSchema.getField(getTopField());
+      if (field == null) {
+        failureCollector.addFailure(
+          String.format("Field '%s' must exist in the input schema.", getTopField()), null)
+          .withConfigProperty(FIELD);
+      } else {
+        Schema fieldSchema = field.getSchema();
+        Schema.Type type = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
+        Schema.LogicalType logicalType = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getLogicalType()
+          : fieldSchema.getLogicalType();
+        if (type != Schema.Type.INT && type != Schema.Type.LONG && type != Schema.Type.FLOAT
+          && type != Schema.Type.DOUBLE || logicalType != null) {
+          failureCollector.addFailure(
+            String.format(
+              "Field '%s' is of type '%s', only types allowed are numeric types.", getTopField(),
+              fieldSchema.getDisplayName()), "Specify field with numeric type.")
```

Comment:
`fieldSchema` should be a non nullable schema. 
```java
Schema nonNullableSchema = fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
Schema.Type type = nonNullableSchema.getType();
Schema.LogicalType logicalType = nonNullableSchema.getLogicalType();
...
String.format(
              "Field '%s' is of type '%s', only types allowed are numeric types.", getTopField(),
              nonNullableSchema.getDisplayName()), "Specify field with numeric type.")
...

Reply:
Done and squashed.